app/src/main/java/com/toshi/view/adapter/ListSectionAdapter.kt

```
@@ -0,0 +1,53 @@
+/*
+ * 	Copyright (c) 2017. Toshi Inc
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     This program is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.toshi.view.adapter
+
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import com.toshi.R
+import com.toshi.view.adapter.viewholder.ListSectionViewHolder
+
+class ListSectionAdapter(
+        private val title: String,
+        private val clickableString: String,
+        private val onSectionClickedListener: () -> Unit
+) : BaseCompoundableAdapter<ListSectionViewHolder, Int>() {
+
+    init {
+        setItemList(listOf(1))
+    }
+
+    override fun compoundableBindViewHolder(viewHolder: RecyclerView.ViewHolder, adapterIndex: Int) {
+        val typedHolder = viewHolder as? ListSectionViewHolder
+                ?: throw AssertionError("This is not the right type!")
```

Wouldn't it be easier and cleaner to just let it throw a ClassCastException instead when the view holder is not what it is supposed to be?
```
val typeHolder = viewHolder as ListSectionViewHolder
```

---

I think @martintoften was following a pattern i started but you're probably right